python/mxnet/initializer.py

```diff
@@ -91,11 +91,37 @@ def _init_weight(self, _, arr):
 
 
 class Xavier(Initializer):
-    """Initialize the weight with Xavier initialization scheme."""
+    """Initialize the weight with Xavier or similar initialization scheme.
+
+    Parameters
+    ----------
+    rnd_type: str, optional
+        Use ```gaussian``` or ```uniform``` to init
+
+    factor_type: str, optional
+        Use ```xavier``` or ```kaiming``` to init
+
+    magnitude: float, optional
+        scale of random number range
+    """
+    def __init__(self, rnd_type="uniform", factor_type="xavier", magnitude=6):
+        self.rnd_type = rnd_type
+        self.factor_type = factor_type
+        self.magnitude = magnitude
+
 
     def _init_weight(self, _, arr):
         shape = arr.shape
         fan_in, fan_out = np.prod(shape[1:]), shape[0]
-        scale = np.sqrt(3. / (fan_in + fan_out))
-        random.uniform(-scale, scale, out=arr)
-
+        factor = 1
+        if self.factor_type == "xavier":
+            factor = fan_in + fan_out
```

Comment:
It doesn't matter in practice. We need to turn it a little bit to avoid too large weight or 0. Default is not always good and lead training fail in many case